crates/bevy_ui/src/flex/convert.rs

```diff
@@ -17,13 +17,25 @@ where
     }
 }
 
-fn from_size<T, U>(size: Size<U>) -> stretch::geometry::Size<T>
+fn from_size<T>(size: Size<Val>) -> stretch::geometry::Size<T>
 where
-    T: From<U>,
+    T: From<Val>,
 {
+    let width = if let Val::Undefined = size.width {
```

Comment:
This will prevent users from overriding size to `Undefined` as well as changing the default behaviour. Perhaps this is not desirable?

Reply:
I actually think we need to keep `Undefined` as an option, so we need should try to find a way to write `impl Default for Size<Val>`. @cart do you have any ideas for how to do this without conflicting with `impl<T> std::default::Default for bevy_math::Size<T>` and also to avoid the `only traits defined in the current crate can be implemented for arbitrary types` error?

Reply:
Yeah this is a little tough. If we could use the nightly "specialization" rust features, we would be able to override the default for this specific generic type. Sadly we can't do that so we'll need to get creative. The current approach seems a little bit hack-ey, but off the top of my head I don't have a nice solution. I suppose we could remove the "default" Size<T> impl and then explicitly implement defaults per-type (ex: Size<Val> , Size<f32>, etc). I'm not a huge fan, but I think I prefer it to the current approach. I guess lets do that for now and we can revisit this later if it comes up.

Reply:
Yes, I think that is probably the best way to solve this.

This fixes the conflict issue, but it doesn't fix the `only traits defined in the current crate can be implemented for arbitrary types` problem. (I think this is a case of the "orphan rule"?)

I am actually surprised that this is an issue, I thought this would have been covered by [the relaxed restrictions on implementing traits in Rust 1.41.0](https://blog.rust-lang.org/2020/01/30/Rust-1.41.0.html#relaxed-restrictions-when-implementing-traits) but I guess not. 😕 ❓

Reply:
We might need to re-define the `Size` struct type locally rather than importing it from `bevy_math`.

Reply:
By the way, I think this bug and the compile errors are both potentially signs of this being implemented in an overly generic way. So I don't mind the idea of explicitly setting all the default values per-type. I think that approach is possibly less error-prone.

Reply:
weird. I was certain that pattern was supported. 

I think the fix here might be to just change the default values for Style.size:

```rust

impl Default for Style {
    fn default() -> Self {
        Self {
            display: Default::default(),
            position_type: Default::default(),
            direction: Default::default(),
            flex_direction: Default::default(),
            flex_wrap: Default::default(),
            align_items: Default::default(),
            align_self: Default::default(),
            align_content: Default::default(),
            justify_content: Default::default(),
            position: Default::default(),
            margin: Default::default(),
            padding: Default::default(),
            border: Default::default(),
            flex_grow: 0.0,
            flex_shrink: 1.0,
            flex_basis: Val::Auto,
            // change to these
            size: Size::new(Val::Auto, Val::Auto),
            min_size: Size::new(Val::Auto, Val::Auto),
            max_size: Size::new(Val::Auto, Val::Auto),
            aspect_ratio: Default::default(),
        }
    }
}

```

Then we can still use the generic type "everywhere". We

Reply:
Ohh of course. That makes sense. Thanks for your help, I’ll update my PR later today.